src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Windows.cs

```diff
@@ -171,11 +171,11 @@ public void RunAsClient(PipeStreamImpersonationWorker impersonationWorker)
             // now handle win32 impersonate/revert specific errors by throwing corresponding exceptions
             if (execHelper._impersonateErrorCode != 0)
             {
-                WinIOError(execHelper._impersonateErrorCode);
+                throw WinIOError(execHelper._impersonateErrorCode);
             }
             else if (execHelper._revertImpersonateErrorCode != 0)
             {
-                WinIOError(execHelper._revertImpersonateErrorCode);
+                throw WinIOError(execHelper._revertImpersonateErrorCode);
```

Comment:
Ouch.

Reply:
Can we add tests that would cause these exceptions to be thrown?

Reply:
Done.